src/userCLI/src/main/kotlin/org/imperial/mrc/hint/userCLI/DatabaseProperties.kt

```diff
@@ -0,0 +1,33 @@
+package org.imperial.mrc.hint.userCLI
+
+import org.imperial.mrc.hint.HintProperties
+import org.imperial.mrc.hint.getResource
+
+open class Environment {
+    open fun get(name: String): String? {
+        return System.getenv(name)
+    }
+}
+
+class DatabaseProperties(private val environment: Environment = Environment()) {
+
+    private val properties = HintProperties().apply {
+        load(getResource("config.properties").openStream())
+    }
+
+    private fun fromEnvOrDefault(propName: String): String {
+        return environment.get(propName) ?: properties[propName].toString()
+    }
+
+    val user: String get() {
+        return fromEnvOrDefault("db_user")
+    }
+
+    val password: String get() {
+        return fromEnvOrDefault("db_password")
```

Comment:
these props are all implicitly under test in here: https://github.com/mrc-ide/hint/pull/144/files#diff-85c6cbe4768ca3720255b3c06ef36a41R17